File: c++代码/637. 二叉树的层平均值.cpp
#include <iostream>
#include <vector>
#include <queue>
using namespace std;

struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution
{
public:
    vector<double> averageOfLevels(TreeNode *root)
    {
        if (!root)
        {
            return ans;
        }

        queue<TreeNode *> q;
        q.push(root);
        while (!q.empty())
        {
            int n = q.size();
            double tmp_val = 0;
            for (int i = 0; i < n; i++)
            {
                TreeNode* node = q.front();
                q.pop();
                tmp_val += node->val;
                if (node->left) q.push(node->left);
                if (node->right) q.push(node->right);
            }
            ans.push_back(tmp_val/n);
        }
        return ans;
    }
    vector<double> ans;
};

File: 力扣/树/637. 二叉树的层平均值.py
# Definition for a binary tree node.]
import queue
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def averageOfLevels(self, root: TreeNode) -> list:
        if not root:
            return []
        q = queue.Queue()
        ans_lst = []
        q.put(root)
        while not q.empty():
            n = q.qsize()
            tmp_val = 0
            for i in range(n):
                node = q.get()
                if node.left:
                    q.put(node.left)
                if node.right:
                    q.put(node.right)
                tmp_val += node.val
            ans_lst.append(tmp_val/n)
        return ans_lst